src/main/java/org/jabref/gui/bibtexextractor/BibtexExtractorViewModel.java

```diff
@@ -12,28 +8,15 @@
 import org.jabref.gui.DialogService;
 import org.jabref.gui.StateManager;
 import org.jabref.gui.externalfiles.ImportHandler;
-import org.jabref.gui.util.BackgroundTask;
 import org.jabref.gui.util.TaskExecutor;
-import org.jabref.logic.importer.FetcherException;
-import org.jabref.logic.importer.fetcher.GrobidCitationFetcher;
-import org.jabref.logic.l10n.Localization;
 import org.jabref.model.database.BibDatabaseContext;
-import org.jabref.model.entry.BibEntry;
 import org.jabref.model.util.FileUpdateMonitor;
 import org.jabref.preferences.PreferencesService;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-public class BibtexExtractorViewModel {
-
-    private static final Logger LOGGER = LoggerFactory.getLogger(BibtexExtractorViewModel.class);
+public abstract class BibtexExtractorViewModel {
```

Comment:
I don't like the idea of using hiearchies here, Favor composition over Inheritance

Reply:
Normal Java practicse, but I can try to redo it. Will be more lines of code!